packages/libp2p/src/connection-manager/auto-dial.ts

```diff
@@ -1,285 +0,0 @@
-    this.autoDialPeerRetryThresholdMs = init.autoDialPeerRetryThreshold ?? defaultOptions.autoDialPeerRetryThreshold
-    this.autoDialDiscoveredPeersDebounce = init.autoDialDiscoveredPeersDebounce ?? defaultOptions.autoDialDiscoveredPeersDebounce
-    this.log = components.logger.forComponent('libp2p:connection-manager:auto-dial')
-    this.started = false
-    this.running = false
-    this.queue = new PeerQueue({
-      concurrency: init.autoDialConcurrency ?? defaultOptions.autoDialConcurrency,
-      metricName: 'libp2p_autodial_queue',
-      metrics: components.metrics
-    })
-    this.queue.addEventListener('error', (evt) => {
-      this.log.error('error during auto-dial', evt.detail)
-    })
-
-    // check the min connection limit whenever a peer disconnects
-    components.events.addEventListener('connection:close', () => {
-      this.autoDial()
-        .catch(err => {
-          this.log.error(err)
-        })
-    })
-
-    // sometimes peers are discovered in quick succession so add a small
-    // debounce to ensure all eligible peers are autodialed
-    let debounce: ReturnType<typeof setTimeout>
-
-    // when new peers are discovered, dial them if we don't have
-    // enough connections
-    components.events.addEventListener('peer:discovery', () => {
-      clearTimeout(debounce)
-      debounce = setTimeout(() => {
-        this.autoDial()
-          .catch(err => {
-            this.log.error(err)
-          })
-      }, this.autoDialDiscoveredPeersDebounce)
-    })
-  }
-
-  isStarted (): boolean {
-    return this.started
-  }
-
-  start (): void {
-    this.started = true
-  }
-
-  afterStart (): void {
-    this.autoDial()
-      .catch(err => {
-        this.log.error('error while autodialing', err)
-      })
-  }
-
-  stop (): void {
-    // clear the queue
-    this.queue.clear()
-    clearTimeout(this.autoDialInterval)
-    this.started = false
-    this.running = false
-  }
-
-  async autoDial (): Promise<void> {
-    if (!this.started || this.running) {
-      return
-    }
-
-    const connections = this.connectionManager.getConnectionsMap()
-    const numConnections = connections.size
-
-    // already have enough connections
-    if (numConnections >= this.minConnections) {
-      if (this.minConnections > 0) {
-        this.log.trace('have enough connections %d/%d', numConnections, this.minConnections)
-      }
-
-      // no need to schedule next autodial as it will be run when on
-      // connection:close event
-      return
-    }
-
-    if (this.queue.size > this.autoDialMaxQueueLength) {
-      this.log('not enough connections %d/%d but auto dial queue is full', numConnections, this.minConnections)
-      this.sheduleNextAutodial()
-      return
-    }
-
-    this.running = true
-
-    this.log('not enough connections %d/%d - will dial peers to increase the number of connections', numConnections, this.minConnections)
```

Comment:
IIRC, we already have logic somewhere that prevents the need to dial bootstrappers if we have highly rated "close" peers in the peer store.. if it's elsewhere other than the auto-dialer, then this can all probably go away.

Question: does the auto-dialer tag/modify peers in the peer store that could change behavior for other services/components? It doesn't look like it does, but I could be missing something.

Reply:
i think the peer-discovery-bootstrap component does this..?

Reply:
> does the auto-dialer tag/modify peers in the peer store

No, it only reads peers from the peer store.

> IIRC, we already have logic somewhere that prevents the need to dial bootstrappers if we have highly rated "close" peers in the peer store

The notion of "close" is down to the protocol that defines it.

For example KAD-DHT already populates it's routing tables with peer store peers on startup, but it doesn't need to actively dial those peers, only during an actual query.

In the case where those peers have gone away between restarts, the KAD paper defines how to ping old peers to make sure they're still contactable during routing table maintenance.